File: src/net/nihlen/bf2/objects/Player.java
package net.nihlen.bf2.objects;

/**
 * A model of a player on the Battlefield 2 server.
 * 
 * @author Alex
 */
public class Player {
	
	public final int index;
	public final String name;
	public final int pid;
	public final String ipAddress;
	public final String hash;
	public String country;
	public String countryCode;
	public int rank;

	public Position position;
	public Rotation rotation;
	public Vehicle rootVehicle;
	public String subVehicle;

	public int teamId;
	public boolean isAlive;

	public int totalScore;
	public int teamScore;
	public int kills;
	public int deaths;
	public int ping;

	private int authLevel;

	public Player(int index, String name, int pid, String ipAddress, String hash, int teamId) {
		this.index = index;
		this.name = name;
		this.pid = pid;
		this.ipAddress = ipAddress;
		this.hash = hash;
		this.teamId = teamId;
		this.isAlive = false;
		this.authLevel = 0;
	}
	
	/*public void delete() {
		if (rootVehicle != null) {
			rootVehicle.removePlayer(this);
		}
	}*/
	
	public boolean isDriver() {
		return ((rootVehicle != null) && (rootVehicle.templateName.equals(subVehicle)));
	}
	
	public synchronized void updateScore(int totalScore, int teamScore, int kills, int deaths) {
		this.totalScore = totalScore;
		this.teamScore = teamScore;
		this.kills = kills;
		this.deaths = deaths;
	}

	public synchronized void setPosition(Position pos) {
		this.position = pos;
	}

	public synchronized void setRotation(Rotation rot) {
		this.rotation = rot;
	}

	public synchronized void setAlive(boolean isAlive) {
		this.isAlive = isAlive;
	}
	
	public synchronized void setTeam(int teamId) {
		this.teamId = teamId;
	}
	
	public synchronized void setVehicle(Vehicle rootVehicle, String subVehicle) {
		this.rootVehicle = rootVehicle;
		this.subVehicle = subVehicle;
	}

	public synchronized void setRank(int rankNum) {
		this.rank = rankNum;
	}

	public int getAuthLevel() {
		return authLevel;
	}
	
	@Override
	public String toString() {
		return name.trim();
	}

}

File: src/BF2WebAdmin.Server/Entities/Game/Player.cs
using System.Net;

namespace BF2WebAdmin.Server.Entities.Game
{
    public class Player : Entity
    {
        public string Name { get; set; }
        public int Index { get; set; }
        public IPAddress IpAddress { get; set; }
        public string Hash { get; set; }
        public Country Country { get; set; }
        public Rank Rank { get; set; }
        public Vehicle RootVehicle { get; set; }
        public Vehicle SubVehicle { get; set; }
        public Team Team { get; set; }
        public bool IsAlive { get; set; }
        public Score Score { get; set; }
    }

    public class Score
    {
        public int Total { get; set; }
        public int Team { get; set; }
        public int Kills { get; set; }
        public int Deaths { get; set; }
        public int Ping { get; set; }
    }

    public class Country
    {
        public string Name { get; set; }
        public string Code { get; set; }
    }
}
